### PR TITLE
Use Poetry instead of Pipenv

### PR DESCRIPTION
**Pull request by sbrunner**
Also:
* I reorganize the `package.json` to respect the names
* Add `package-lock.json`
* Use less global installation with npm

### CODE REVIEW

File: geoportal/c2cgeoportal_geoportal/scaffolds/advance_update/{{cookiecutter.project}}/geoportal/CONST_Makefile [hunk 1]
```diff
@@ -43,32 +43,12 @@ VALIDATE_PY_FOLDERS = admin/$(PACKAGE)_admin \
...
 
 .PHONY: build
 build: $(BUILD_RULES)
 
-.PHONY: checks
-checks: prospector eslint
-
-.PHONY: prospector
-prospector:
-	prospector
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/advance_update/{{cookiecutter.project}}/geoportal/CONST_Makefile:62]: Do we still have a target to run those checks locally.

**Review by arnaud-morvan — COMMENTED** [review]
Here are some comments after a first review.

File: package.json [hunk 2]
```diff
@@ -4,9 +4,13 @@
   "description": "GeoMapFish geoportal interface",
   "author": "",
   "dependencies": {
+    "ngeo": "2.7.0-version-2.7-latest.20220524T070933Z.0c7639891.2.7",
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply package.json:7]: Yes, it should be possible :-), I will try

[the remark this answers is not in the document]
Comment (sbrunner) [reply package.json:7]: Should we keep package.json or geoportal/package.json?

[the remark this answers is not in the document]
Comment (arnaud-morvan) [reply package.json:7]: We could keep the exiting one in /geoportal
Note that we have another in /admin

File: .prospector.yaml [hunk 3]
```diff
@@ -8,13 +8,12 @@ ignore-paths:
   - setup.py
   - CONST_create_template
   - scaffolds
+  - docker/qgisserver
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply .prospector.yaml:11]: We do the prospector check for QGIS in a separate run (make prospector-qgisserver)

File: Dockerfile [hunk 4]
```diff
@@ -29,96 +52,78 @@ RUN \
...
-RUN python3 -m pip install --disable-pip-version-check --no-cache-dir --requirement=/tmp/requirements.txt \
-    && rm --recursive --force /tmp/*
-
-COPY Pipfile Pipfile.lock /tmp/
-# hadolint disable=DL3003
-RUN cd /tmp && PIP_NO_BINARY=fiona,rasterio,shapely PROJ_DIR=/usr/local/ pipenv sync --system --clear \
-    && rm --recursive --force /usr/local/lib/python3.*/dist-packages/tests/ /tmp/* /root/.cache/* \
+RUN --mount=type=cache,target=/root/.cache \
+    --mount=type=bind,from=poetry,source=/tmp,target=/poetry \
+    IP_NO_BINARY=fiona,rasterio,shapely PROJ_DIR=/usr/local/ python3 -m pip install --disable-pip-version-check --no-deps --requirement=/poetry/requirements.txt \
```
Comment (arnaud-morvan) [inline Dockerfile:59]:
```suggestion
    PIP_NO_BINARY=fiona,rasterio,shapely PROJ_DIR=/usr/local/ python3 -m pip install --disable-pip-version-check --no-deps --requirement=/poetry/requirements.txt \
```

File: Makefile [hunk 5]
```diff
@@ -122,3 +146,22 @@ doc: build-tools ## Generate the documentation
...
+	python3 -m pip install --user transifex-client
+
+admin/c2cgeoportal_admin/locale/%/LC_MESSAGES/c2cgeoportal_admin.mo: transifex
+	rm /tmp/c2ctemplate-cache.json
+	make --makefile=build.mk $@
+
+ADMIN_PO_FILES = admin/c2cgeoportal_admin/locale/fr/LC_MESSAGES/c2cgeoportal_admin.mo
+.PHONY: dev
+dev: $(ADMIN_PO_FILES) ## Generate a development environment that can be mount in a project container
+	(cd admin; npm install)
```
Comment (arnaud-morvan) [inline Makefile:162]: Not sure to understand why we need this.
Admin JS packages might be installed in /opt in Docker images.

File: bin/build-l10n [hunk 6]
```diff
@@ -7,7 +7,7 @@ import shutil
 import subprocess
 
 
-def main():
+def _main() -> None:
```
Comment (arnaud-morvan) [inline bin/build-l10n:10]: Wreid, you want the main function to be protected, or you need the main symbol for something else ?

File: ci/test-new-project [hunk 7]
```diff
@@ -36,10 +37,14 @@ initial_access_length = 0
 
 url = sys.argv[1]
 
-# See also:
-# ci/vars.yaml geoportal/GUNICORN_PARAMS
-# Jenkinsfile timeout line 130
-result = requests.get(url, timeout=240, verify=False)  # nosec
+for _ in range(10):
+    print(f"Qeuery {url}")
```
Comment (arnaud-morvan) [inline ci/test-new-project:41]:
```suggestion
    print(f"Query {url}")
```

File: geoportal/c2cgeoportal_geoportal/scaffolds/create/{{cookiecutter.project}}/Makefile [hunk 8]
```diff
@@ -2,12 +2,35 @@ PROJECT_PUBLIC_URL=https://example.camptocamp.com/
...
+help: ## Display this help message
+	@echo "Usage: make <target>"
+	@echo
+	@echo "Available targets:"
+	@grep --extended-regexp --no-filename '^[a-zA-Z_-]+:.*## ' $(MAKEFILE_LIST) | sort | \
+		awk 'BEGIN {FS = ":.*?## "}; {printf "	%-20s%s\n", $$1, $$2}'
+
 .PHONY: update-po-from-url
-update-po-from-url:
+update-po-from-url: ## Updtate the po files from the URL provide by PROJECT_PUBLIC_URL
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/create/{{cookiecutter.project}}/Makefile:14]:
```suggestion
update-po-from-url: ## Update the po files from the URL provide by PROJECT_PUBLIC_URL
```

**Review by arnaud-morvan — APPROVED** [review]

File: Makefile (hunk 5 above)
Comment (sbrunner) [reply Makefile:162]: Effectively, removed :-)

**Pull request merged**